engine/src/engine.ts

```diff
@@ -1165,7 +1166,7 @@ export default class Engine {
 
   [Phase.SetupFaction](move: string) {
     this.loadTurnMoves(move, { split: false, processFirst: true });
-    if (this.isVersionOrLater("5.0.0")) {
+    if (this.isVersionOrLater("4.7.0")) {
```

Comment:
Last nitpick: I think this can be `!this.isOnLegacyAuction()` as well

Reply:
As far as I understand the website correctly:

- It loads JSON data from MongoDB.
- It calls the `move(data, _move, player)` function in `engine/wrapper.ts`.
- It restores the engine based on the JSON.

If there is a game being started while the update rolls, it ends up being in a hybrid state: e.g. the first faction is chosen old style and the second is chosen new style. The game then does not proceed to the next phase after the last player picked their faction. I added a test that verifies the behavior. It fails when I change this line to `!this.isOnLegacyAuction()`.

Reply:
Good catch :)